sympy/solvers/ode.py

```diff
@@ -5573,6 +5611,67 @@ def _solve_variation_of_parameters(eq, func, order, match):
         psol = trigsimp(psol, deep=True)
     return Eq(f(x), gsol.rhs + psol)
 
+def ode_factorable(eq, func, order, match):
+    r"""
+    Solves equations having a solvable factor.
+
+    This function is used to solve the equation having factors. Factors may be of type algebraic or ode. It
+    will try to solve each factor independently. ODE type of factors will be solved by calling dsolve and algebraic type
+    of factors will be solved by calling solve. We will return the list of solutions.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, dsolve, Eq, pprint, Derivative
+    >>> from sympy.abc import x
+    >>> f = Function('f')
+    >>> eq = x**3*f(x)**2 + x**3*f(x)*Derivative(f(x), (x, 2)) + x**2*f(x)*Derivative(f(x), x) - 4*x*f(x)**2
+    >>> pprint(dsolve(eq, f(x)))
+                  /     2\
+                2 |    x |    / 6\
+    [f(x) = C1*x *|1 - --| + O\x /, f(x) = 0]
+                  \    12/
+
+    """
+    x = func.args[0]
+    sols_diff = []
+    sols_algebraic = []
+    for i in match['eqs_diff']:
+        sols=dsolve(i, func)
+        if isinstance(sols, list):
+            for sol in sols:
+                sols_diff.append(sol)
+        else:
+            sols_diff.append(sols)
+
+    # We are solving algebraic equation by calling solve and If We will get any error
+    # from solve We will ignore it(We are here to solve differential equation).
+    for i in match['eqs_algebraic']:
+        try:
+            sols = solve(i, func)
+            for sol in sols:
+                sols_algebraic+=[Eq(func, sol)]
+        except Exception:
+            continue
+
+    # I am removing the series solution from the solution because We can't check redundant of a solution in series solution.
+    new_sols = []
+    for i in range(len(sols_diff)):
+        if isinstance(sols_diff[i], Eq):
+            if sols_diff[i].has(Order):
+                new_sols.append(sols_diff[i])
+                del sols_diff[i]
+
+    # I am using already implemented function to remove reduntant solution if ODE solution exits
+    if len(sols_diff):
+        new_sols += _nth_algebraic_remove_redundant_solutions(eq, sols_diff+sols_algebraic, order, x)
+    else:
+        new_sols += sols_algebraic
+
+    if len(new_sols)==1:
+        return new_sols[0]
+    else:
+        return new_sols
```

Comment:
I can't remember: is this needed? It should really be handled higher up at the end of dsolve.

Reply:
I don't know I will take a look.

Reply:
Not needed